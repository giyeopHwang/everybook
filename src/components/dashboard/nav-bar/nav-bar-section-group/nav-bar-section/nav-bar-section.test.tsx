import { render, screen, fireEvent } from '@testing-library/react';
import NavBarSection from './nav-bar-section.component';

describe('NavBarSection', () => {
  it('renders NavBarSection component', () => {
    const navBarSection = render(
      <NavBarSection title="TITLE" onToggle={() => {}}>
        CHILDREN
      </NavBarSection>
    );
    expect(navBarSection).toMatchSnapshot();

    const title = screen.getByText('TITLE');
    expect(title).toBeInTheDocument();
  });

  it('renders with subTitle when specified', () => {
    const navBarSection = render(
      <NavBarSection title="TITLE" subTitle="SUBTITLE" onToggle={() => {}}>
        CHILDREN
      </NavBarSection>
    );
    expect(navBarSection).toMatchSnapshot();

    const subTitle = screen.getByText('SUBTITLE');
    expect(subTitle).toBeInTheDocument();
  });

  it('renders collapsed when specified', () => {
    const navBarSection = render(
      <NavBarSection title="TITLE" onToggle={() => {}} expanded={false}>
        CHILDREN
      </NavBarSection>
    );
    expect(navBarSection).toMatchSnapshot();
  });

  it('fires callback when title clicked', () => {
    const mock = vi.fn();
    const navBarSection = render(
      <NavBarSection title="TITLE" onToggle={mock}>
        CHILDREN
      </NavBarSection>
    );
    expect(navBarSection).toMatchSnapshot();

    const title = screen.getByText('TITLE');
    fireEvent.click(title);
    expect(mock).toHaveBeenCalledOnce();
  });
});
