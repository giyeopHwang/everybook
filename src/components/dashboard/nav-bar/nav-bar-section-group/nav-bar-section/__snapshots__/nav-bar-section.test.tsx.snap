// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBarSection > fires callback when title clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="css-14chwgd"
      >
        <header
          class="css-kympkc"
        >
          <div
            class="css-359auk"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 1rem; height: 1rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m4.5 15.75 7.5-7.5 7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h2
              class="css-1bt39ed"
            >
              TITLE
            </h2>
          </div>
        </header>
        CHILDREN
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="css-14chwgd"
    >
      <header
        class="css-kympkc"
      >
        <div
          class="css-359auk"
        >
          <svg
            aria-hidden="true"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            style="width: 1rem; height: 1rem;"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <h2
            class="css-1bt39ed"
          >
            TITLE
          </h2>
        </div>
      </header>
      CHILDREN
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavBarSection > renders NavBarSection component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="css-14chwgd"
      >
        <header
          class="css-kympkc"
        >
          <div
            class="css-359auk"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 1rem; height: 1rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m4.5 15.75 7.5-7.5 7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h2
              class="css-1bt39ed"
            >
              TITLE
            </h2>
          </div>
        </header>
        CHILDREN
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="css-14chwgd"
    >
      <header
        class="css-kympkc"
      >
        <div
          class="css-359auk"
        >
          <svg
            aria-hidden="true"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            style="width: 1rem; height: 1rem;"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <h2
            class="css-1bt39ed"
          >
            TITLE
          </h2>
        </div>
      </header>
      CHILDREN
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavBarSection > renders collapsed when specified 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="css-14chwgd"
      >
        <header
          class="css-kympkc"
        >
          <div
            class="css-359auk"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 1rem; height: 1rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m19.5 8.25-7.5 7.5-7.5-7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h2
              class="css-1bt39ed"
            >
              TITLE
            </h2>
          </div>
        </header>
        CHILDREN
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="css-14chwgd"
    >
      <header
        class="css-kympkc"
      >
        <div
          class="css-359auk"
        >
          <svg
            aria-hidden="true"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            style="width: 1rem; height: 1rem;"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m19.5 8.25-7.5 7.5-7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <h2
            class="css-1bt39ed"
          >
            TITLE
          </h2>
        </div>
      </header>
      CHILDREN
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavBarSection > renders with subTitle when specified 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="css-14chwgd"
      >
        <header
          class="css-kympkc"
        >
          <div
            class="css-359auk"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 1rem; height: 1rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m4.5 15.75 7.5-7.5 7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h2
              class="css-1bt39ed"
            >
              TITLE
            </h2>
          </div>
          <h3
            class="css-1puzo0z"
          >
            SUBTITLE
          </h3>
        </header>
        CHILDREN
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="css-14chwgd"
    >
      <header
        class="css-kympkc"
      >
        <div
          class="css-359auk"
        >
          <svg
            aria-hidden="true"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            style="width: 1rem; height: 1rem;"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <h2
            class="css-1bt39ed"
          >
            TITLE
          </h2>
        </div>
        <h3
          class="css-1puzo0z"
        >
          SUBTITLE
        </h3>
      </header>
      CHILDREN
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
